File: server/models/user_model.js
const { Schema, model } = require('mongoose');
const uuid = require('uuid');
const bcrypt = require('bcrypt');
const userSchema = new Schema({
    id: {
        type: String,
        unique: true,

    },
    fullName: {
        type: String,
        default: ""
    },
    email: {
        type: String,
        unique: true,
        required: true
    },
    password: {
        type: String,
        required: true
    },
    phoneNumber: {
        type: String,
        default: ""
    },
    address: {
        type: String,
        default: ""
    },
    city: {
        type: String,
        default: ""
    },
    state: {
        type: String,
        default: ""
    },
    profileProgress: {
        type: Number,
        default: 0,

    },
    updateOn: {
        type: Date,
        default: Date
    },
    createOn: {
        type: Date,
        default: Date
    }
});
userSchema.pre("save", function (next) {
    const user = this;
    this.id = uuid.v1();
    this.createOn = new Date();
    this.updateOn = new Date();

    //store password with hash
    //  
    const saltData = bcrypt.genSaltSync(10);

    const hashPassword = bcrypt.hashSync(this.password, saltData);
    this.password = hashPassword;
    next();
});
userSchema.pre(["update", "findOneAndUpdate", "update"], (next) => {
    const update = this.getUpdate();
    delete update._id;
    delete update.id;
    this.updateOn = new Date();
    next();
});

const UserModel = model("User", userSchema);
module.exports = UserModel;

File: lib/data/models/user/user_model.dart
class UserModel {
  dynamic sId;
  dynamic fullName;
  dynamic email;
  dynamic phoneNumber;
  dynamic address;
  dynamic city;
  dynamic state;
  dynamic profileProgress;
  dynamic id;
  dynamic updatedOn;
  dynamic createdOn;
  UserModel({
    this.sId,
    this.fullName,
    this.email,
    this.phoneNumber,
    this.address,
    this.city,
    this.state,
    this.profileProgress,
    this.id,
    this.createdOn,
    this.updatedOn,
  });
  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      sId: json['_id'],
      fullName: json['fullName'],
      email: json['email'],
      phoneNumber: json['phoneNumber'],
      address: json['address'],
      city: json['city'],
      state: json['state'],
      profileProgress: json['profileProgress'],
      id: json['id'],
      createdOn: json['createOn'],
      updatedOn: json['updateOn'],
    );
  }
}
